packages/backend/src/nest/storage/certificates/certificates.store.ts

```diff
@@ -0,0 +1,138 @@
+import { getCrypto } from 'pkijs'
+
+import { EventEmitter } from 'events'
+import { StorageEvents } from '../storage.types'
+
+import EventStore from 'orbit-db-eventstore'
+import OrbitDB from 'orbit-db'
+
+import { loadCertificate, keyFromCertificate } from '@quiet/identity'
+
+import { ConnectionProcessInfo, NoCryptoEngineError, SocketActionTypes } from '@quiet/types'
+
+import { IsNotEmpty, IsBase64, validate } from 'class-validator'
+import { ValidationError } from '@nestjs/common'
+
+import createLogger from '../../common/logger'
+
+const logger = createLogger('CertificatesStore')
+
+class UserCertificateData {
+    @IsNotEmpty()
+    @IsBase64()
+    certificate: string
+}
+
+export class CertificatesStore {
+    public orbitDb: OrbitDB
+    public store: EventStore<string>
+
+    constructor(orbitDb: OrbitDB) {
+        this.orbitDb = orbitDb
+    }
+
+    public async init(emitter: EventEmitter) {
+        logger('Initializing certificates log store')
+
+        this.store = await this.orbitDb.log<string>('certificates', {
+            replicate: false,
+            accessController: {
+                write: ['*'],
+            },
+        })
+
+        this.store.events.on('write', async (_address, entry) => {
+            logger('Saved certificate locally')
+
+            emitter.emit(StorageEvents.LOAD_CERTIFICATES, {
+                certificates: await this.getCertificates(),
+            })
+
+            // await this.updatePeersList()
+        })
+
+        this.store.events.on('ready', async () => {
```

Comment:
Can we get rid of this event as it probably duplicates what's being done on `db.load`?

Reply:
Yeah, if we do the same on 'ready' and on other event that calls "db.load" then yes, there is a duplication because db.load triggers 'ready'

Reply:
Ok, now it emits `emitter.emit(SocketActionTypes.CONNECTION_PROCESS_INFO, ConnectionProcessInfo.LOADED_CERTIFICATES)` but doesn't trigger certificates loading